openml/study/functions.py

```diff
@@ -1,49 +1,84 @@
+from typing import cast, Dict, List, Optional, Union
+
+import dateutil.parser
 import xmltodict
 
-from openml.study import OpenMLStudy
+from openml.study import OpenMLStudy, OpenMLBenchmarkSuite
+from openml.study.study import BaseStudy
 import openml._api_calls
 
 
-def get_study(study_id, entity_type=None):
+def get_suite(suite_id: Union[int, str]) -> OpenMLBenchmarkSuite:
     """
-    Retrieves all relevant information of an OpenML study from the server
-    Note that some of the (data, tasks, flows, setups) fields can be empty
-    (depending on information on the server)
+    Retrieves all relevant information of an OpenML benchmarking suite from the server.
 
     Parameters
     ----------
     study id : int, str
         study id (numeric or alias)
 
     entity_type : str (optional)
-        Which entity type to return. Either {data, tasks, flows, setups,
-        runs}. Give None to return all entity types.
+        Do not change this!
+
+    Returns
+    -------
+    OpenMLSuite
+        The OpenML suite object
+    """
+    suite = cast(OpenMLBenchmarkSuite, _get_study(suite_id, entity_type='tasks'))
+    return suite
+
+
+def get_study(
+    study_id: Union[int, str],
+    arg_for_backwards_compat: Optional[str] = None,
+) -> OpenMLStudy:  # noqa F401
+    """
+    Retrieves all relevant information of an OpenML study from the server.
+
+    Parameters
+    ----------
+    study id : int, str
+        study id (numeric or alias)
 
     Returns
     -------
     OpenMLStudy
         The OpenML study object
     """
-    call_suffix = "study/%s" % str(study_id)
-    if entity_type is not None:
-        call_suffix += "/" + entity_type
+    if study_id == 'OpenML100':
+        study = _get_study(study_id, entity_type='tasks')
+        return cast(OpenMLBenchmarkSuite, study)  # type: ignore
```

Comment:
Users here are (almost definitely) doing this because they read the outdated OpenML100 paper.
What do you think of outputting a warning specifying something along the lines of
"It looks like you are running code from the OpenML100 paper. It still works, but lots of things have changed since then. Please visit <url> for an up-to-date example." or something? The page could then also highlight some of the most important differences and/or link to documentation (e.g. concept of benchmark vs. suite).